processor/src/main/resources/org/mapstruct/ap/internal/model/StreamMappingMethod.ftl

```diff
@@ -52,7 +52,7 @@
     <#assign needVarDefine = beforeMappingReferencesWithMappingTarget?has_content && !existingInstanceMapping />
 
     <#if resultType.arrayType>
-        <#if !existingInstanceMapping && needVarDefine>
```

Comment:
the `!existinInstanceMapping` is already present in `needVarDefine`, hence it could be removed.